src/custom_nodes/model_zoo_intel_object_detection/model_zoo_intel_object_detection.cpp

```diff
@@ -183,40 +183,55 @@ int initializeInternalManager(void** customNodeLibraryInternalManager, const str
     NODE_ASSERT(targetImageHeight > 0, "target image height must be larger than 0");
     NODE_ASSERT(targetImageWidth > 0, "target image width must be larger than 0");
 
+    bool status = true;
     // creating BuffersQueues for output: images
     int channels = convertToGrayScale ? 1 : 3;
     uint64_t imagesByteSize = sizeof(float) * targetImageHeight * targetImageWidth * channels * maxOutputBatch;
     imagesByteSize = imagesByteSize;
-    internalManager->createBuffersQueue(OUTPUT_IMAGES_TENSOR_NAME, imagesByteSize, QUEUE_SIZE);
-    internalManager->createBuffersQueue(OUTPUT_IMAGES_DIMS_NAME, 5 * sizeof(uint64_t), QUEUE_SIZE);
+    status = internalManager->createBuffersQueue(OUTPUT_IMAGES_TENSOR_NAME, imagesByteSize, QUEUE_SIZE);
```

Comment:
Instead of using temp variable, you can simply call:
```
if (!internalManager->createBuffersQueue(OUTPUT_IMAGES_TENSOR_NAME, imagesByteSize, QUEUE_SIZE))
  return 1;
```

What is more, should we clean up buffers when initialization fails? (return 1)

Reply:
In pipelinedefinition.cpp PipelineDefinition::initializeNodeResources() we create `void* customNodeLibraryInternalManager = nullptr;` and then in initialize() manager is creating buffers. In case of failure  currently we are immediately returning StatusCode and do nothing with allocated manager and its members(not saving into nodeResources).

I see two solutions. One would be actually saving created manager even tho it failed. Than during reload it would try to initialize again and potentially success.

The other one would be to call `delete internalManager`. With that solution internalManager and its members would be destroyed and in pipelinedefinition pointer would not be saved.

Reply:
See https://github.com/openvinotoolkit/model_server/blob/07c406d2cafdc55c12b323c067962837dddb3547/src/pipelinedefinition.cpp#L90